File: C#/Circular/Circular.cs
using System;

public class Node
{
    public int Data;
    public Node Next;

    public Node(int data)
    {
        Data = data;
        Next = null;
    }
}

public class CircularLinkedList
{
    public Node Head;

    public CircularLinkedList()
    {
        Head = null;
    }

    public void Insert(int data)
    {
        Node newNode = new Node(data);
        if (Head == null)
        {
            Head = newNode;
            newNode.Next = Head;
        }
        else
        {
            Node temp = Head;
            while (temp.Next != Head)
            {
                temp = temp.Next;
            }
            temp.Next = newNode;
            newNode.Next = Head;
        }
    }

    public void DeleteNode(int key)
    {
        if (Head == null) return;

        Node current = Head, prev = null;

        // If the node to be deleted is the head node
        if (current.Data == key)
        {
            while (current.Next != Head)
            {
                current = current.Next;
            }

            if (Head.Next == Head)
            {
                Head = null;
                return;
            }

            current.Next = Head.Next;
            Head = current.Next;
            return;
        }

        // Search for the node to be deleted
        do
        {
            prev = current;
            current = current.Next;
        } while (current != Head && current.Data != key);

        if (current.Data == key)
        {
            prev.Next = current.Next;
        }
    }

    public void Display()
    {
        if (Head == null) return;
        Node current = Head;
        do
        {
            Console.Write(current.Data + " -> ");
            current = current.Next;
        } while (current != Head);
        Console.WriteLine("(head)");
    }
}

class Program
{
    static void Main()
    {
        CircularLinkedList list = new CircularLinkedList();
        list.Insert(3);
        list.Insert(2);
        list.Insert(1);

        Console.WriteLine("Original List:");
        list.Display();

        list.DeleteNode(2);
        Console.WriteLine("After Deleting 2:");
        list.Display();
    }
}

File: JavaScript/Circular/Circular.js
class Node {
    constructor(data) {
        this.data = data;
        this.next = null;
    }
}

class CircularLinkedList {
    constructor() {
        this.head = null;
    }

    insert(data) {
        const newNode = new Node(data);
        if (this.head === null) {
            this.head = newNode;
            newNode.next = this.head;
        } else {
            let temp = this.head;
            while (temp.next !== this.head) {
                temp = temp.next;
            }
            temp.next = newNode;
            newNode.next = this.head;
        }
    }

    deleteNode(key) {
        if (this.head === null) return;

        let current = this.head;
        let prev = null;

        // If the node to be deleted is the head node
        if (current.data === key) {
            while (current.next !== this.head) {
                current = current.next;
            }

            if (this.head.next === this.head) {
                this.head = null;
                return;
            }

            current.next = this.head.next;
            this.head = current.next;
            return;
        }

        // Search for the node to be deleted
        do {
            prev = current;
            current = current.next;
        } while (current !== this.head && current.data !== key);

        if (current.data === key) {
            prev.next = current.next;
        }
    }

    display() {
        if (this.head === null) return;
        let current = this.head;
        do {
            process.stdout.write(current.data + ' -> ');
            current = current.next;
        } while (current !== this.head);
        console.log('(head)');
    }
}

// Example usage
const list = new CircularLinkedList();
list.insert(3);
list.insert(2);
list.insert(1);

console.log('Original List:');
list.display();

list.deleteNode(2);
console.log('After Deleting 2:');
list.display();
